components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementPolicyAdminService.java

```diff
@@ -369,21 +361,18 @@ public void removePolicy(String policyId, boolean dePromote) throws EntitlementE
         }
         handleStatus(EntitlementConstants.StatusTypes.DELETE_POLICY, oldPolicy, true, null);
 
-        //remove versions
-        EntitlementAdminEngine.getInstance().getVersionManager().deletePolicy(policyId);
```

Comment:
Is this change intentional? @Niru-D

Reply:
Yes, removing policy versions is done inside the policy removal method implementation of the new interface.